Description: In Text::Template 0.44 _scrubpkg is still causing memory leaks
 in long running processes. The generated package is not cleared and the
 package is not removed from the symbol table of Text::Template.
Origin: https://rt.cpan.org/Public/Bug/Display.html?id=22031
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=22031
Bug-Debian: http://bugs.debian.org/609528
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=22031
Author: Radu Greab <radu [...] yx.ro>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-01-10

--- a/lib/Text/Template.pm
+++ b/lib/Text/Template.pm
@@ -428,6 +428,8 @@
     foreach my $key (keys %$hash) {
       undef $hash->{$key};
     }
+    %$hash = ();
+    delete $Text::Template::{$s."::"};
   }
 }
   
--- a/t/01-basic.t
+++ b/t/01-basic.t
@@ -245,7 +245,9 @@
 # (31) Test _scrubpkg for leakiness
 $Text::Template::GEN0::test = 1;
 Text::Template::_scrubpkg('Text::Template::GEN0');
-if ($Text::Template::GEN0::test) {
+if ($Text::Template::GEN0::test
+    || exists $Text::Template::GEN0::{test}
+    || exists $Text::Template::{'GEN0::'}) {
   print "not ok $n\n";
 } else {
   print "ok $n\n";
